java-frontend/src/main/java/org/sonar/java/se/checks/CustomUnclosedResourcesCheck.java

```diff
@@ -40,17 +41,28 @@
 
 import javax.annotation.Nullable;
 
+import java.util.List;
+
 @Rule(key = "S3546")
 @RuleTemplate
 public class CustomUnclosedResourcesCheck extends SECheck {
 
-  static class ResourceStatus implements ObjectConstraint.Status {
+  public class CustomResourceConstraint implements Constraint {
```

Comment:
Probably worth mentioning that it's also related to https://jira.sonarsource.com/browse/SONARJAVA-1624